File: product-service/src/main/java/com/github/buyalsky/productservice/dto/ProductStockChangeDto.java
package com.github.buyalsky.productservice.dto;

import java.util.Objects;

public class ProductStockChangeDto {
    private String productId;
    private Integer currentLeftInStock;

    public ProductStockChangeDto() {
    }

    public ProductStockChangeDto(String productId, Integer currentLeftInStock) {
        this.productId = productId;
        this.currentLeftInStock = currentLeftInStock;
    }

    public String getProductId() {
        return productId;
    }

    public Integer getCurrentLeftInStock() {
        return currentLeftInStock;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    public void setCurrentLeftInStock(Integer currentLeftInStock) {
        this.currentLeftInStock = currentLeftInStock;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ProductStockChangeDto that = (ProductStockChangeDto) o;
        return productId.equals(that.productId) && currentLeftInStock.equals(that.currentLeftInStock);
    }

    @Override
    public int hashCode() {
        return Objects.hash(productId, currentLeftInStock);
    }
}

File: shopping-cart-service/src/main/kotlin/com/github/buyalsky/shoppingcartservice/dto/ProductStockChangeDto.kt
package com.github.buyalsky.shoppingcartservice.dto

import com.github.buyalsky.shoppingcartservice.dto.ProductStockChangeDto
import java.util.*

class ProductStockChangeDto {
    var productId: String? = null
    var currentLeftInStock: Int? = null

    constructor() {}
    constructor(productId: String?, currentLeftInStock: Int?) {
        this.productId = productId
        this.currentLeftInStock = currentLeftInStock
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ProductStockChangeDto
        return productId == that.productId && currentLeftInStock == that.currentLeftInStock
    }

    override fun hashCode(): Int {
        return Objects.hash(productId, currentLeftInStock)
    }
}
